### PR TITLE
fix: remove unnecessary type conversion in MysqlWorker

### LINKED ISSUE #596: Remove unnecessary type conversion in MysqlWorker
### Describe This Problem

When I'm implement #578, I find query in MysqlWorker return type used for JSON, which is not necessary, and may cost more resources.

### Proposal

Remove this conversion
https://github.com/CeresDB/ceresdb/blob/43a70fb4fa3462eeb760ba5b671c708e20a22aa4/server/src/mysql/worker.rs#L119

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Huachao**
# Which issue does this PR close?

Closes #596 

# Rationale for this change
 

# What changes are included in this PR?

Remove `Output` to `Response` convension

# Are there any user-facing changes?

No

# How does this change test

CI

### CODE REVIEW

**Comment by jiacai2050** [comment]
@Huachao Thanks for your contribution.

Our CI doesn't cover MySQL protocols now, would you mind add one?  or you just can test by yourself, and paste what you tested here.

File: server/src/mysql/writer.rs [hunk 1]
```diff
@@ -36,21 +35,49 @@ impl<'a, W: std::io::Write> MysqlQueryResultWriter<'a, W> {
...
+        let mut column_data = vec![];
+
+        for record_batch in records {
+            let num_cols = record_batch.num_columns();
+            let num_rows = record_batch.num_rows();
+            let schema = record_batch.schema();
+
+            for col_idx in 0..num_cols {
+                let column_schema = schema.column(col_idx).clone();
+                column_names.push(ResponseColumn {
```
[the remark this answers is not in the document]
Comment (Huachao) [reply server/src/mysql/writer.rs:55]: > 1. If there are more than more records, will column_names include duplicated column names?

I am not entirely familiar with `RecordBatch`, can we treat each `RecordBatch` in the vector has the same column names and different data? If so I think I can change it to only keep the column names in the first loop iteration.

> 2. Also, I think we can convert our column directly to `Column` in opensrv, no need to convert to ResponseColumn.

Nice suggestion, additionally I'd like to remove the convert here, just keep the `column_schema`,      and do the conversion in line 76 where we need to write.

What's your opinion, @jiacai2050 Thanks in advance

**Comment by Huachao** [comment]
@jiacai2050 currently I test it manually with following sql statements.

1. Create a `test` table
```sql
CREATE TABLE `test` (`name` string TAG, `value` double NOT NULL, `t` timestamp NOT NULL, TIMESTAMP KEY(t)) ENGINE=Analytic with (enable_ttl='false');
```
2. Check the empty `test` table output of `select` query
```bash
mysql> select * from test;
Query OK, 0 rows affected (0.05 sec)
```
3. Insert the first record
```sql
insert into test (t, name, value) VALUES (1651737067000, 'ceresdb', 100);
```
4. Check the `test` table output of `select` query
```bash
mysql> select * from test;
+----------------------+---------------+---------+-------+
| tsid                 | t             | name    | value |
+----------------------+---------------+---------+-------+
| 12128845460635970325 | 1651737067000 | ceresdb |   100 |
+----------------------+---------------+---------+-------+
1 row in set (0.00 sec)
```
5. Insert the second record
```sql
insert into test (t, name, value) VALUES (1651737068000, 'ceresdb', 200);
```
6. Check the `test` table output of `select` query
```bash
mysql> select * from test;
+----------------------+---------------+---------+-------+
| tsid                 | t             | name    | value |
+----------------------+---------------+---------+-------+
| 12128845460635970325 | 1651737067000 | ceresdb |   100 |
| 12128845460635970325 | 1651737068000 | ceresdb |   200 |
+----------------------+---------------+---------+-------+
2 rows in set (0.00 sec)
```

File: server/src/mysql/writer.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiacai2050) [reply server/src/mysql/writer.rs]: > I am not entirely familiar with RecordBatch, can we treat each RecordBatch in the vector has the same column names and different data? If so I think I can change it to only keep the column names in the first loop iteration.

Yes, schema in RecordBatch vector is the same, you can fetch column name using the first one.

**Comment by jiacai2050** [comment]
The tests look good 👍

We can merge this PR after you fix column name extraction using only first RecordBatch.

**Comment by jiacai2050** [comment]
@Huachao  I have push minor updates to your branch,  will merge when CI pass.

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Comment by jiacai2050** [comment]
Thanks again 🚀

**Pull request merged**